pages/dao/[symbol]/proposal/components/instructions/PsyFinance/MintAmericanOptions.tsx

```diff
@@ -0,0 +1,319 @@
+import { PublicKey, TransactionInstruction } from '@solana/web3.js'
+import {
+  Governance,
+  ProgramAccount,
+  serializeInstructionToBase64,
+} from '@solana/spl-governance'
+import React, { useContext, useEffect, useReducer } from 'react'
+import Input from '@components/inputs/Input'
+import Tooltip from '@components/Tooltip'
+import useGovernanceAssets from '@hooks/useGovernanceAssets'
+import { AssetAccount } from '@utils/uiTypes/assets'
+import {
+  PsyFinanceMintAmericanOptionsForm,
+  UiInstruction,
+} from '@utils/uiTypes/proposalCreationTypes'
+import { useState } from 'react'
+import {
+  ASSOCIATED_TOKEN_PROGRAM_ID,
+  MintInfo,
+  Token,
+  TOKEN_PROGRAM_ID,
+} from '@solana/spl-token'
+import { BN } from 'bn.js'
+import {
+  createProgram,
+  deriveOptionKeyFromParams,
+  getOptionByKey,
+  instructions,
+} from '@mithraic-labs/psy-american'
+import BigNumber from 'bignumber.js'
+import { tryGetMint } from '../../../../../../../utils/tokens'
+import { NewProposalContext } from '../../../new'
+import GovernedAccountSelect from '../../GovernedAccountSelect'
+import { PSY_AMERICAN_PROGRAM_ID } from '../../../../../../../utils/instructions/PsyFinance'
+import useWallet from '../../../../../../../hooks/useWallet'
+import { getATA } from '../../../../../../../utils/ataTools'
+
+const formReducer = (
+  state: PsyFinanceMintAmericanOptionsForm,
+  action: Partial<PsyFinanceMintAmericanOptionsForm>
+) => ({
+  ...state,
+  ...action,
+})
+
+const MintAmericanOptions = ({
+  index,
+  governance,
+}: {
+  index: number
+  governance: ProgramAccount<Governance> | null
+}) => {
+  const { anchorProvider, connection, wallet } = useWallet()
+  const { governedTokenAccountsWithoutNfts } = useGovernanceAssets()
+  const { handleSetInstructions } = useContext(NewProposalContext)
+  const [underlyingMintInfo, setUnderlyingMintInfo] = useState<MintInfo | null>(
+    null
+  )
+  const [form, dispatch] = useReducer(formReducer, {
+    contractSize: 1,
+    expirationUnixTimestamp: 0,
+    optionTokenDestinationAccount: '',
+    quoteMint: '',
+    size: 0,
+    strike: 0,
+    underlyingAccount: undefined,
+    underlyingMint: undefined,
+    writerTokenDestinationAccount: '',
+  })
+
+  const shouldBeGoverned = !!(index !== 0 && governance)
+
+  const getInstruction = async (): Promise<UiInstruction> => {
+    if (
+      !underlyingMintInfo ||
+      !form.quoteMint ||
+      !form.strike ||
```

Comment:
Is `0` a valid value for `strike`?

Reply:
It's valid at the protocol level. However, a strike of 0 would be the same as giving the underlying for free so it doesn't make sense at the user level.